tests/fields/test_decimal.py

```diff
@@ -95,6 +95,53 @@ def test_decimal_places_bad(self):
             values[0], {"sum_decimal": Decimal("37.26")},
         )
 
+    async def test_aggregate_sum_with_f_expression(self):
+        await testmodels.DecimalFields.create(decimal=Decimal("0"), decimal_nodec=1)
+        await testmodels.DecimalFields.create(decimal=Decimal("9.99"), decimal_nodec=1)
+        await testmodels.DecimalFields.create(decimal=Decimal("27.27"), decimal_nodec=1)
+        values = (
+            await testmodels.DecimalFields.all()
+            .annotate(sum_decimal=Sum(F("decimal")))
+            .values("sum_decimal")
+        )
+        self.assertEqual(
+            values[0], {"sum_decimal": Decimal("37.26")},
+        )
+
+        values = (
+            await testmodels.DecimalFields.all()
+            .annotate(sum_decimal=Sum(F("decimal") + 1))
+            .values("sum_decimal")
+        )
+        self.assertEqual(
+            values[0], {"sum_decimal": Decimal("40.26")},
+        )
+
+        values = (
+            await testmodels.DecimalFields.all()
+            .annotate(sum_decimal=Sum(F("decimal") + F("decimal")))
+            .values("sum_decimal")
+        )
+        self.assertEqual(
+            values[0], {"sum_decimal": Decimal("74.52")},
+        )
+
+    async def test_aggregate_sum_fail_with_f_expression(self):
+        with self.assertRaisesRegex(
+            FieldError,
+            "Field not_exist is not default field in model, so can not be used with F expression",
+        ):
+            await testmodels.DecimalFields.all().annotate(sum_decimal=Sum(F("not_exist"))).values(
+                "sum_decimal"
+            )
+
+        with self.assertRaisesRegex(
+            FieldError, "Cannot use arithmetic expression between different field type"
```

Comment:
Could you split these tests? There is no reason to have them both in the same test?

Reply:
Sure. I will split those.